geode-core/src/integrationTest/java/org/apache/geode/cache/query/JoinQueriesIntegrationTest.java

```diff
@@ -120,4 +175,167 @@ private void populateRegionWithData(Region region1, Region region2) {
+
+    results = (SelectResults) queryService.newQuery(queryString).execute();
+    int resultsSizeWithIndex = results.size();
+    assertThat(resultsWithoutIndex).isEqualTo(expectedResultSize);
+    assertThat(resultsWithoutIndex).isEqualTo(resultsSizeWithIndex);
+  }
+
+  private void populateTripleJointRegions(int expectedMatches, int extraEntitiesPerRegion,
+      Cache cache, Region<String, Object> orderRegion, Region<String, Object> validationIssueRegion,
+      Region<String, Object> validationIssueXRefRegion) {
+    for (int i = 0; i < expectedMatches; i++) {
+      PdxInstance orderPdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.Order")
+          .writeString("orderId", "orderId_" + i)
+          .writeInt("version", i)
+          .create();
+      PdxInstance validationIssuePdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.ValidationIssue")
+          .writeString("issueId", "issueId_" + i)
+          .writeDate("createdTime", Calendar.getInstance().getTime())
+          .create();
+      PdxInstance validationIssueXRefPdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.OrderValidationIssueXRef")
+          .writeString("validationIssueXRefID", "validationIssueXRefID_" + i)
+          .writeString("referenceIssueId", "issueId_" + i)
+          .writeString("referenceOrderId", "orderId_" + i)
+          .writeInt("referenceOrderVersion", i)
+          .create();
+
+      orderRegion.put("orderId_" + i, orderPdx);
+      validationIssueRegion.put("issueId_" + i, validationIssuePdx);
+      validationIssueXRefRegion.put("validationIssueXRefID_" + i, validationIssueXRefPdx);
+    }
+
+    for (int i = 0; i < extraEntitiesPerRegion; i++) {
+      PdxInstance orderPdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.Order")
+          .writeString("orderId", "orderId#" + i)
+          .writeInt("version", i)
+          .create();
+      PdxInstance validationIssuePdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.ValidationIssue")
+          .writeString("referenceIssueId", "referenceIssueId#" + i)
+          .writeDate("createdTime", Calendar.getInstance().getTime())
+          .create();
+      PdxInstance validationIssueXRefPdx = cache
+          .createPdxInstanceFactory("org.apache.geode.test.OrderValidationIssueXRef")
+          .writeString("validationIssueXRefID", "validationIssueXRefID#" + i)
+          .writeString("referenceIssueId", "validationIssueID2#" + i)
+          .writeString("referenceOrderId", "orderId#2" + i)
+          .writeInt("referenceOrderVersion", i)
+          .create();
+
+      orderRegion.put("orderId#" + i, orderPdx);
+      validationIssueRegion.put("issueId#" + i, validationIssuePdx);
+      validationIssueXRefRegion.put("validationIssueXRefID#" + i, validationIssueXRefPdx);
+    }
+  }
+
+  private void populateTripleJointRegions(int expectedMatches, int extraEntitiesPerRegion,
```

Comment:
Nitpick: Is the difference here object types vs the previous method putting pdx?  If so, would it make sense to rename the other one to populateTripleJointRegionsWithPDX or pass in some supplier to create the objects to be put?